apps/aecore/lib/aecore/peers/peer_connection.ex

```diff
@@ -529,27 +530,148 @@ defmodule Aecore.Peers.PeerConnection do
     ]
   end
 
-  def rlp_encode(@ping, %{
-        share: share,
-        genesis_hash: genesis_hash,
-        best_hash: best_hash,
-        difficulty: difficulty,
-        peers: peers,
-        port: port
-      }) do
+  # RLP for peer messages
+
+  defp rlp_encode(@p2p_response, %{result: result, type: type, object: object, reason: reason}) do
+    serialized_result = bool_bin(result)
+
+    reason =
+      case reason do
+        nil ->
+          <<>>
+
+        reason ->
+          reason
+      end
+
+    serialized_object =
+      case object do
+        nil ->
+          <<>>
+
+        object ->
+          rlp_encode(type, object)
+      end
+
+    list = [
+      @p2p_msg_version,
+      serialized_result,
+      :binary.encode_unsigned(type),
+      reason,
+      serialized_object
+    ]
+
+    ExRLP.encode(list)
```

Comment:
I think the variable list here and in all the other `rlp_encode()` functions is redundant. Just pass the list as the argument. In some of the functions it is already done in this way, why not do it for all of them?